runtime/codert_vm/cnathelp.cpp

```diff
@@ -3013,7 +3013,9 @@ old_slow_jitNewInstanceImplAccessCheck(J9VMThread *currentThread)
 				J9UTF8_LENGTH(nameUTF),
 				J9UTF8_DATA(sigUTF),
 				J9UTF8_LENGTH(sigUTF));
-		setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALACCESSEXCEPTION, detailMessage);
+		if (NULL != detailMessage) {
```

Comment:
Is it better to throw an empty IAE here than the NativeOOM?

Reply:
I don't like the idea of setting two exceptions in a row, but I could be talked out of it. It won't actually cause any problems to do so.

Reply:
I'm OK with throwing the nativeOOM as that's the immediate failure.  Either way, the user will need to resolve both the cause of the OOM and the IAE - given this is a native OOM, trying to create a second exception (and its stacktrace) to set one as the cause of the other is asking for trouble.

Let's keep it as you've written it